### PR TITLE
Bug 1849368 - Remove data path and data authorities requirement for open links in apps

### PR DESCRIPTION
**Pull request by rocketsroger**
On non-pixel devices, data path count and data authorities count requirements filters out external applications that can open the link.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1849368

### CODE REVIEW

**Comment by Djfe** [comment]
I'm experiencing Bug 1849368 on a Pixel, how weird

**Comment by Djfe** [comment]
I got `target.arm64-v8a.apk` from the `firefoxci-taskcluster` as described above and installed it on my Pixel 7 Pro (Android 13 Stock ROM, September 23 Security Update).
I was able to verify that the .apk above has working
`mailto:`, `tel:`, `sms:` and `geo:` links again :)

I also retested current Release (117.1.0) and Nightly after my last OS update (September security release) and they are still affected.

**Comment by rocketsroger** [comment]
> I got `target.arm64-v8a.apk` from the `firefoxci-taskcluster` as described above and installed it on my Pixel 7 Pro (Android 13 Stock ROM, September 23 Security Update). I was able to verify that the .apk above has working `mailto:`, `tel:`, `sms:` and `geo:` links again :)

Thanks for confirming the fix will work.

File: android-components/components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt [hunk 1]
```diff
@@ -183,10 +183,9 @@ class AppLinksUseCases(
...
                     // that can open this app link
                     ANDROID_RESOLVER_PACKAGE_NAME, null -> {
-                        findActivities(appIntent).filter {
-                            it.filter != null &&
-                                !(it.filter.countDataPaths() == 0 && it.filter.countDataAuthorities() == 0)
-                        }.getOrNull(0)
+                        findActivities(appIntent).firstOrNull {
+                            it.filter != null
+                        }
                     }
```
Comment (Amejia481) [inline android-components/components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt:189]: Is it possible to write tests to verify this new behaviour?  Or the existing ones already covers it?

Comment (rocketsroger) [reply android-components/components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt:189]: I'll check to see if there's a good test for this.  Thanks

Comment (rocketsroger) [reply android-components/components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt:189]: @Amejia481 I couldn't find anything good test to add here.  Existing tests already covered what we need.

Comment (Amejia481) [reply android-components/components/feature/app-links/src/main/java/mozilla/components/feature/app/links/AppLinksUseCases.kt:189]: Thanks for the clarification :)

**Review by Amejia481 — APPROVED** [review]

**Comment by rocketsroger** [comment]
@Mergifyio backport releases_v119

**Comment by rocketsroger** [comment]
@Mergifyio backport releases_v118

**Comment by rocketsroger** [comment]
backing out the change in Main since a contributor have a very similar change.

**Pull request merged**